fmrest/cloudserver.py

```diff
@@ -0,0 +1,132 @@
+                 cognito_client_id: str = '4l9rvl4mv5es1eep1qe97cautn',
+                 data_sources: Optional[List[Dict]] = None,
+                 verify_ssl: Union[bool, str] = True,
+                 type_conversion: bool = False,
+                 auto_relogin: bool = False
+                 ) -> None:
+        """Initialize the CloudServer class.
+
+        Parameters
+        ----------
+        url : str
+            Address of the FileMaker Cloud Server, e.g. https://myteamname.account.filemaker-cloud.com
+            Note: Data API must use https.
+        user : str
+            Claris ID name to log into your database
+            Note: make sure it belongs to a privilege set that has fmrest extended privileges.
+        password : str
+            Claris ID Password to log into your database
+        database : str
+            Name of database without extension, e.g. Contacts
+        layout : str
+            Layout to work with. Can be changed between calls by setting the layout attribute again,
+            e.g.: fmrest_instance.layout = 'new_layout'.
+        cognito_userpool_id: str
+            Amazon Cognito Userpool ID (required for FileMaker Cloud authentication).
+            See https://help.claris.com/en/customer-console-help/content/create-fmid-token.html
+            At time of writing, this value should be 'us-west-2_NqkuZcXQY'
+        cognito_client_id:
+            Amazon Cognito Client ID (required for FileMaker Cloud authentication).
+            See https://help.claris.com/en/customer-console-help/content/create-fmid-token.html
+            At time of writing, this value should be '4l9rvl4mv5es1eep1qe97cautn'
+        data_sources : list, optional
+            List of dicts in format
+                [{'database': 'db_file', 'username': 'admin', 'password': 'admin'}]
+            Use this if for your actions you need to be authenticated to multiple DB files.
+        verify_ssl : bool or str, optional
+            Switch to set if certificate should be verified.
+            Use False to disable verification. Default True.
+            Use string path to a root cert pem file, if you work with a custom CA.
+        type_conversion : bool, optional
+            If True, attempt to convert string values into their potential original types.
+            In previous versions of the FileMaker Data API only strings were returned and there was
+            no way of knowing the correct type of a requested field value.
+
+            Be cautious with this parameter, as results may be different from what you expect!
+
+            Values will be converted into int, float, datetime, timedelta, string. This happens
+            on a record level, not on a foundset level.
+        auto_relogin : bool, optional
+            If True, tries to automatically get a new token (re-login) when a
+            request comes back with a 952 (invalid token) error. Defaults to
+            False.
+        """
+
```

Comment:
As mentioned above, I would here check for the availability of `pycognito`:

```
        if not _has_pycognito:
            raise ImportError(
                'Please install pycognito for Claris Cloud support. '
                'You can do so with: pip install python-fmrest[cloud]'
            )
```

Reply:
This is a great idea! Thank you for this. I will add it.